packages/panzer/disc-fe/src/Panzer_PointDescriptor.hpp

```diff
@@ -0,0 +1,137 @@
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Roger P. Pawlowski (rppawlo@sandia.gov) and
+// Eric C. Cyr (eccyr@sandia.gov)
+// ***********************************************************************
+// @HEADER
+
+
+#ifndef __Panzer_PointDescriptor_hpp__
+#define __Panzer_PointDescriptor_hpp__
+
+#include <string>
+
+#include "Teuchos_RCP.hpp"
+
+// forward declarations
+namespace shards {
+class CellTopology;
+}
+
+namespace panzer {
+class PointGenerator;
+}
+
+// class declarations
+namespace panzer {
+
+class PointDescriptor
+{
+public:
+ 
+  /** \brief Constructor for the point descriptor. Protected and used by constructors.
+    *
+    * \param[in] type String that defines the "type" of this point descriptor,
+    *                 used to generate unique hashes
+    * \param[in] generator PointGenerator object for the points.
+    */
+  PointDescriptor(const std::string & type,const Teuchos::RCP<PointGenerator> & generator);
+
+  /// Destructor
+  virtual ~PointDescriptor() = default;
+
+  /** Build a generator class that generates any reference points on 
+    * a specified topology.
+    *
+    * \param[in] The cell topology to build the coordinates on
+    */
+  const PointGenerator & getGenerator() const { return *_generator; }
+
+  /** \brief Get unique string associated with the type of point
+   *         descriptor. This will be used generate a hash
+   *        to sort through a map of PointDescriptors.
+   *
+   * \return  A string that uniquely describes this point descriptor
+   */
+  const std::string & getType() const { return _type; }
+
+  /** \brief Get unique key associated with integrator of this order and type
+   *  The key is used to sort through a map of IntegrationDescriptors.
+   *
+   * \return Unique basis key
+   */
+  std::size_t getKey() const {return _key;}
+
+protected:
+
+  /// Default constructor, no user version
+  PointDescriptor() {}
```

Comment:
Yea,
```cpp
public:
  PointDescriptor() = delete;
```
means no default constructor will exist, neither from you or from auto-generation by the compiler.